core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -397,10 +405,10 @@ protected Node(final Environment environment, Collection<Class<? extends Plugin>
                     b.bind(HttpServer.class).toProvider(Providers.of(null));
                 };
             }
-            final DiscoveryModule discoveryModule = new DiscoveryModule(this.settings, transportService, networkService,
+
+            final DiscoveryModule discoveryModule = new DiscoveryModule(this.settings, threadPool, transportService,
+                networkService, clusterService, hostsProvider -> newZenPing(settings, threadPool, transportService, hostsProvider),
```

Comment:
should we replace newZenPing with a MockZenDiscovery? I *think* this would allow removing ZenPing from the `getDiscoveryTypes` interface, replacing it with a UnicastHostPrivider. Might be simpler?